stacker/lookups/handlers/dynamodb.py

```diff
@@ -74,35 +56,100 @@ def handler(value, **kwargs):
     else:
         raise ValueError(
-            'No dynamo record with those paramters could be found')
+            'The dynamodb record could not be found using the following '
+            'key: {}'.format(new_keys[0]))
+
+
+def _lookup_key_parse(table_keys):
+    """Return the order in which the stacks should be executed.
+
+    Args:
+        - dependencies (dict): a dictionary where each key should be the
+            fully qualified name of a stack whose value is an array of
+            fully qualified stack names that the stack depends on. This is
+            used to generate the order in which the stacks should be
+            executed.
+
+    Returns:
+        dict: includes a dict of lookup types with data types ('new_keys')
+              and a list of the lookups with without ('clean_table_keys')
 
+    """
+    # we need to parse the key lookup passed in
+    regex_matcher = '\[([^\]]+)]'
+    valid_dynamodb_datatypes = ['M', 'S', 'N', 'L']
+    clean_table_keys = []
+    new_keys = []
 
-def _buildProjectionExpression(clean_table_keys):
-    """this builds the lookup for dynamodb
+    for key in table_keys:
+        match = re.search(regex_matcher, key)
+        if match:
+            # the datatypes are pulled from the dynamodb docs
+            if match.group(1) in valid_dynamodb_datatypes:
+                match_val = str(match.group(1))
+                key = key.replace(match.group(0), '')
+                new_keys.append({match_val: key})
+                clean_table_keys.append(key)
+            else:
+                raise ValueError(
+                    ('Stacker does not support looking up the datatype: {}')
+                    .format(str(match.group(1))))
+        else:
+            new_keys.append({'S': key})
+            clean_table_keys.append(key)
+    key_dict = {}
+    key_dict['new_keys'] = new_keys
+    key_dict['clean_table_keys'] = clean_table_keys
+
+    return key_dict
+
+
+def _build_projection_expression(clean_table_keys):
+    """Given cleaned up keys, this will return a projection expression for
+    the dynamodb lookup.
+
+    Args:
+        clean_table_keys (dictionary): keys without the data types attached
+
+    Returns:
+        string: A projection expression for the dynamodb lookup.
     """
-    projection_expression = ""
+    projection_expression = ''
     for key in clean_table_keys[:-1]:
-        projection_expression += ("{},").format(key)
+        projection_expression += ('{},').format(key)
     projection_expression += clean_table_keys[-1]
     return projection_expression
 
 
-def _getValFromDict(data, keylist):
-    """This finds the key inside the data returned by dynamodb
+def _get_val_from_ddb_data(data, keylist):
+    """Given a dictionary of dynamodb data (including the datatypes) and a
+    properly structured keylist, it will return the value of the lookup
+
+    Args:
+        - data(dictionary): the raw dynamodb data
```

Comment:
no `- ` at the beginning, change `dictionary` -> `dict`, also you're missing a space between the name and the type